src/main/java/cubicchunks/server/chunkio/IONbtReader.java

```diff
@@ -178,6 +178,9 @@ private static void readEntities(NBTTagCompound nbt, ICubicWorldServer world, Cu
 			}
 
 			entity.addedToChunk = true;
+			entity.chunkCoordX = cube.getX();
```

Comment:
The entity ticking code is kind of hard to understand... So there is more than just one myth that could be created (World.updateEntityWithOptionalForce(boolean forcedUpdate) is even misnamed)
I will add a comment that says something like `// this entity is in a Cube right? so it needs to know the cube coords`